botocore/args.py

```diff
@@ -224,42 +197,6 @@ def compute_s3_config(self, scoped_config, client_config):
 
         return s3_configuration
 
-    def _compute_endpoint_config(self, service_name, region_name, endpoint_url,
-                                 is_secure, endpoint_bridge):
-        endpoint_config = endpoint_bridge.resolve(
-            service_name, region_name, endpoint_url, is_secure)
-        if self._should_set_global_sts_endpoint(
-                service_name, region_name, endpoint_url):
-            self._set_global_sts_endpoint(endpoint_config, is_secure)
-        return endpoint_config
-
-    def _should_set_global_sts_endpoint(self, service_name, region_name,
-                                        endpoint_url):
-        if service_name != 'sts':
-            return False
-        if endpoint_url:
-            return False
-        return (
-            self._get_sts_regional_endpoints_config() == 'legacy' and
-            region_name in LEGACY_GLOBAL_STS_REGIONS
-        )
-
-    def _get_sts_regional_endpoints_config(self):
-        sts_regional_endpoints_config = self._config_store.get_config_variable(
-            'sts_regional_endpoints')
-        if not sts_regional_endpoints_config:
-            sts_regional_endpoints_config = 'legacy'
-        if sts_regional_endpoints_config not in \
-                VALID_STS_REGIONAL_ENDPOINTS_CONFIG:
-            raise botocore.exceptions.InvalidSTSRegionalEndpointsConfigError(
```

Comment:
We should remove `InvalidSTSRegionalEndpointsConfigError` from `exceptions.py` as well.